### PR TITLE
add Matomo headers to 11ty and jsdoc; closes #3785

### LINKED ISSUE #3785: add matomo headers to doc layouts
It will need to go in the 11ty layout *and* the API layout.

I don't want to have to update `@mocha/docdash` to make the latter happen; hopefully it allows you to add a random header via configuration.

### PR DESCRIPTION
**Pull request by boneskull**
- use new `headScripts` config per `@mocha/docdash` @ 2.1.0
- do not send analytics in dev environment
- add subdomain tracking

once this is published, then must remove headers from Netlify UI.

### CODE REVIEW

**Comment by Alhadis** [comment]
> `@mocha/docdash`

This should have been named "Docha". :wink:

File: docs/_includes/default.html [hunk 1]
```diff
@@ -16,6 +16,8 @@
     <link rel="shortcut icon" href="favicon.ico" />
 
     <!--[if lt IE 9]> <script src="js/html5shiv.min.js"></script> <![endif]-->
+
+    <script src="/js/matomo.js"></script>
```
Comment (plroebuck) [inline docs/_includes/default.html]: Why begin `src` attribute URL with slash?

File: jsdoc.conf.json [hunk 2]
```diff
@@ -5,7 +5,8 @@
   },
   "mocha-docdash": {
     "sort": true,
-    "static": false
+    "static": false,
+    "headScripts": ["/js/matomo.js"]
```
Comment (plroebuck) [inline jsdoc.conf.json]: same question from above

Comment (boneskull) [reply jsdoc.conf.json]: This is necessary because these are in `/api/`, and the js is in `/js/`, not `/api/js/`.

File: docs/js/matomo.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
+// don't trigger analytics for local development environments
+if (document.domain !== 'localhost') {
```
Comment (outsideris) [inline docs/js/matomo.js]: Shouldn't we prevent the trigger for netlify preview like https://deploy-preview-3789--mocha.netlify.com/ ?

Only trigger analytics in production site like:

```js
if  (document.domain.indexOf('mochajs.org') !== -1)
```

Comment (boneskull) [reply docs/js/matomo.js]: good idea

**Comment by JoshuaKGoldberg** [comment]
Closing out to keep our queue small. Someone please yell at me if this should be re-opened though. Cheers! 🤎

**Pull request closed without merging**